example/kotlinlib/basic/1-simple/build.mill

```diff
@@ -0,0 +1,101 @@
+//// SNIPPET:BUILD
+package build
+import mill._, kotlinlib._, scalalib._
+
+object `package` extends RootModule with KotlinModule {
+
+  def kotlinVersion = "1.9.24"
+
+  def mainClass = Some("foo.FooKt")
+
+  def ivyDeps = super.ivyDeps() ++ Agg(
+    ivy"net.sourceforge.argparse4j:argparse4j:0.9.0",
+    ivy"org.apache.commons:commons-text:1.12.0"
+  )
+
+  object test extends KotlinModuleTests with TestModule.Junit4{
+    def ivyDeps = super.ivyDeps() ++ Agg(
+      ivy"com.google.guava:guava:33.3.0-jre"
+    )
+  }
```

Comment:
For the purposes of these examples, let's swap over to Kotlin libs where possible:

* https://github.com/ajalt/clikt seems to be a popular argument parser and can replace `argparse4j`
* https://github.com/Kotlin/kotlinx.html can replace `commons-text` for what we're doing (we use Scalatags in the Scala versions)
* https://github.com/kotest/kotest seems like the winner for Kotlin-specific test frameworks
* If we use `kotest` via `ivyDeps`, we can drop the `guava` thing (whose main use in the Java example is to show test third party dependencies, and which we do not have in the Scala example)

Reply:
I would say that plenty of tests in Kotlin-first codebases are using JUnit framework, but for the sake of the examples/illustrations/sanity check `kotest` can be used indeed.